tests/storage_test.go

```diff
@@ -0,0 +1,260 @@
+	"github.com/stretchr/testify/assert"
+)
+
+var db *sql.DB
+var dbStorage *storage.DB
+var ctx context.Context
+
+func TestMain(m *testing.M) {
+	// make sure the DB is cleaned of all rows
+	ctx = context.Background()
+	ctx = logevent.NewContext(ctx, stdoutLogger())
+	source, err := settings.NewEnvSource(os.Environ())
+	if err != nil {
+		panic(err.Error())
+	}
+
+	postgresConfigComponent := &storage.PostgresConfigComponent{}
+	dbStorage = new(storage.DB)
+	if err = settings.NewComponent(ctx, source, postgresConfigComponent, dbStorage); err != nil {
+		panic(err.Error())
+	}
+
+	db, err = connectToDB()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// if err = dropTables(db); err != nil {
+	// 	panic(err.Error())
+	// }
+
+	// if err = createTables(db); err != nil {
+	// 	panic(err.Error())
+	// }
+
+	os.Exit(m.Run())
+}
+
+func TestDoIt(t *testing.T) {
+
+	before(t, db)
+
+	if err := dbStorage.StoreCloudAsset(ctx, fakeCloudAssetChanges()); err != nil {
+		panic(err)
+	}
+
+	from, _ := time.Parse(time.RFC3339, "2019-08-08T08:29:35+00:00")
+	to, _ := time.Parse(time.RFC3339, "2019-08-10T08:29:35+00:00")
+	networkChangeEvents, err := dbStorage.GetIPAddressesForTimeRange(ctx, from, to)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, networkChangeEvent := range networkChangeEvents {
+		json, _ := json.Marshal(networkChangeEvent)
+		fmt.Printf(string(json))
+	}
+
+}
+
+func TestAAA(t *testing.T) {
+
+	before(t, db)
+
+	// code should tolerate no data in the tables
+
+	from, _ := time.Parse(time.RFC3339, "2019-08-08T08:29:35+00:00")
+	to, _ := time.Parse(time.RFC3339, "2019-08-10T08:29:35+00:00")
+	networkChangeEvents, err := dbStorage.GetIPAddressesForTimeRange(ctx, from, to)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	assert.Equal(t, 0, len(networkChangeEvents), "there really should have been zero rows returned")
+
+}
+
+// returns a raw sql.DB object, rather than the storage.DB abstraction, so
+// we can perform some Postgres cleanup/prep that are test-specific
+func connectToDB() (*sql.DB, error) {
+	host := os.Getenv("POSTGRES_HOSTNAME")
+	port := os.Getenv("POSTGRES_PORT")
+	user := os.Getenv("POSTGRES_USERNAME")
+	password := os.Getenv("POSTGRES_PASSWORD")
+	dbname := os.Getenv("POSTGRES_DATABASENAME")
+
+	sslmode := "disable"
+	if host != "localhost" && host != "postgres" {
+		sslmode = "require"
+	}
+	psqlInfo := fmt.Sprintf("host=%s port=%s user=%s "+
+		"password=%s dbname=%s sslmode=%s",
+		host, port, user, password, dbname, sslmode)
+	pgdb, err := sql.Open("postgres", psqlInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	err = pgdb.Ping()
+	if err != nil {
+		return nil, err
+	}
+
+	return pgdb, nil
+}
+
+func before(t *testing.T, db *sql.DB) {
+	if err := dropTables(db); err != nil {
+		t.Fatalf("Failed to DROP tables due to: %s", err.Error())
+	}
+	if err := createTables(db); err != nil {
+		t.Fatalf("Failed to CREATE tables due to: %s", err.Error())
+	}
+}
+
+func dropTables(db *sql.DB) error {
+
+	sqlFile := "1_clean.sql"
```

Comment:
I don't see this file

Reply:
It's been checked in now, in a squashed commit.